web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -28,57 +28,79 @@ class ProfileEditForm extends FormBase implements ContainerInjectionInterface {
   /**
    * The DBCDK Community Service Profile API.
    *
-   * @var ProfileApi $profileApi
+   * @var \DBCDK\CommunityServices\Api\ProfileApi $profileApi
    */
   protected $profileApi;
 
   /**
    * The Community Service Profile object from the username context.
    *
-   * @var Profile $profile
+   * @var \DBCDK\CommunityServices\Model\Profile $profile
    */
   protected $profile;
 
   /**
+   * Array of all defined community roles.
+   *
+   * @var \DBCDK\CommunityServices\Model\CommunityRole[] $communityRoles
+   */
+  protected $communityRoles;
+
+  /**
    * Creates a Profile Edit Form instance.
    *
-   * @param null|Profile $profile
-   *   The current Community Profile or NULL if we had no successful matches.
-   * @param ProfileApi $profile_api
+   * @param \DBCDK\CommunityServices\Model\Profile[] $profile
+   *   Results array with a single Community Profile element.
+   * @param \DBCDK\CommunityServices\Model\CommunityRole[] $community_roles
+   *   Results array with all possible Community Roles.
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
    *   The DBCDK Community Service Profile API.
    */
-  public function __construct($profile, ProfileApi $profile_api) {
+  public function __construct(array $profile, array $community_roles, ProfileApi $profile_api) {
     $this->profileApi = $profile_api;
-    // Set the profile as NULL if we did not get any results array from the
-    // service - or get the first element of the array, since we only get one
-    // result, and set this as our profile property.
-    $this->profile = empty($profile) ? NULL : $profile[0];
+    $this->communityRoles = $community_roles;
+    $this->profile = $profile[0];
   }
 
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
-    // Get the Profile API and fetch the Profile we wish to edit and inject
-    // this into the forms constructor.
     $profile_api = $container->get('dbcdk_community.api.profile');
+    $community_roles_api = $container->get('dbcdk_community.api.community_roles');
+
     try {
+      // Get the profile we wish to alter with the form.
       $filter = [
         'limit' => 1,
         'where' => [
           'username' => $container->get('request_stack')->getCurrentRequest()->get('username'),
         ],
       ];
-      /* @var Profile $profile */
-      $profile = $profile_api->profileFind(json_encode($filter));
+      /* @var \DBCDK\CommunityServices\Model\Profile[] $profile */
+      $profile = (array) $profile_api->profileFind(json_encode($filter));
```

Comment:
How about using [`profileFindOne`](https://github.com/DBCDK/biblo-admin/blob/develop/lib/client/src/Api/ProfileApi.php#L796) instead? This will only return one `Profile ` instance.

If not then we should reduce the return value of `profileFind` to a single object as soon as possible to avoid checking and passing around arrays later on.